Actions/ReadSecrets/ReadSecretsHelper.psm1

```diff
@@ -62,17 +62,17 @@ function Get-KeyVaultCredentials {
         [switch] $dontmask
     )
     if ($script:isKeyvaultSet) {
+        $jsonStr = $script:gitHuBSecrets.AZURE_CREDENTIALS
+        if ($jsonStr -contains "`n" -or $jsonStr -contains "`r") {
+            throw "Secret AZURE_CREDENTIALS cannot contain line breaks"
```

Comment:
Why not simply ```$jsonStr -replace "`n","" -replace "`r",""```?

I suppose you tried something like it already, just it's always worth asking.

Reply:
Having linebreaks in the secrets have caused the output to show parts of the secret.
This construct:

${{ secrets.something }}

Is pre-compiled and with newlines it becomes invalidated yaml and an error with exposed partial secrets.

It might be that we are handling these things better today and are not exposed to that - but I don't see any reason for the structured secrets to include newlines - they should be compressed JSON.